src/services/ComponentLoader.tsx

```diff
@@ -2,75 +2,210 @@ import React from 'react';
 
 import ContentTitle from '../components/content/ContentTitle';
 import ContentImage from '../components/content/ContentImage';
-import ContentValue from '../components/content/ContentValue';
 import ContentLink from '../components/content/ContentLink';
 import ContentKeyValue from '../components/content/ContentKeyValue';
 import ContentDivider from '../components/content/ContentDivider';
+import { ComponentConfig, Entity, Value, LeafComponentConfig, FormatterConfig } from '../typings/schema';
+import { valueToString } from '../services/getValue';
 
-import { COMPONENTS } from '../constants/global';
-import { ComponentValue } from '../typings/schema';
-
-interface ComponentLoaderProps {
-    component: any;
-    data: any;
+interface TypedUri { 
+    uri: string; 
+    type: string;
 }
+type uriOrString = string | TypedUri;
 
-const ComponentLoader = ({ component, data }: ComponentLoaderProps) => {
+type pathResult = uriOrString[] | uriOrString | undefined;
 
-    const getTreeValue = (fields) => {
-        let tree = { ...data };
+// `type: never` makes the type checker report an error if the case switch does not handle all types
+function checkUnknownComponent(type: never) {
+    console.error(`Type ${(type as ComponentConfig).type} is not handled!`);
+}
 
-        for (let i = 0, len = fields.length; i < len; i++) {
-            const key = fields[i];
+function isValue(obj: Value | Entity): obj is Value {
+    return obj.hasOwnProperty('value');
+}
 
-            if (!tree || !tree[key]) {
-                break;
-            }
+function walkPath(pathStr: string | undefined, formatters: FormatterConfig, entity: Entity): pathResult {
+    if (pathStr) {
+        const path = pathStr.split('.').map(segment => segment.split(':')[0]);
+        
+        return walkPathStep(path, formatters, entity);
+    }
+    return undefined;
+}
 
-            tree = tree[key];
+function walkPathStep(path: string[], formatters: FormatterConfig, entity: Entity): pathResult {
+    if (path.length === 0) {
+        return entity.uri;
+    }
+    let result = entity[path[0]];
+
+    if (!result) {
+        return undefined;
+    } else if (Array.isArray(result)) {
+        let retVal: uriOrString[] = [];
+        for (const item of result) {
+            if (isValue(item)) {
+                retVal.push(valueToString(item, formatters.concat([{
+                    type: 'http://timbuctoo.huygens.knaw.nl/datatypes/person-name',
```

Comment:
Should this be an addition to the formatter + can you always assume that if an item has a value, it's exactly this: 
`{ type: 'http://timbuctoo.huygens.knaw.nl/datatypes/person-name', name: 'PERSON_NAMES' } `?

If so, I'd recommend to also put this in a const, because the same value is used again below.